choco-solver/src/main/java/org/chocosolver/solver/Solver.java

```diff
@@ -873,8 +889,15 @@ public void findOptimalSolution(ResolutionPolicy policy, RealVar objective, doub
         if (!getObjectiveManager().isOptimization()) {
             set(new ObjectiveManager<RealVar, Double>(objective, policy, precision, true));
         }
-        set(new LastSolutionRecorder(new Solution(), true, this));
-        solve(false);
+        ISolutionRecorder oldRecorder = getSolutionRecorder();
```

Comment:
Well, if one manually declared a solution recorder, it does not seem a good idea to me to unplug it just before running the search.
Is the idea behind the patch to avoid using the wrong solution recorder when switching from optimization to satisfaction?